src/perception/tag_detector/tag_detector.processing.cpp

```diff
@@ -112,10 +113,18 @@ void FiducialsNode::imageCallback(sensor_msgs::ImageConstPtr const& msg) {
  * @param v     Y Pixel Position
  */
 std::optional<SE3> getFidInCamFromPixel(PointCloudPtr const& cloudPtr, size_t u, size_t v) {
+    if (u >= cloudPtr->width || v >= cloudPtr->height) {
+        ROS_WARN("Tag center out of bounds");
+        return std::nullopt;
+    }
+    
     pcl::PointXYZRGBNormal const& point = cloudPtr->at(static_cast<int>(u), static_cast<int>(v));
-    return (std::isfinite(point.x) && std::isfinite(point.y) && std::isfinite(point.z))
-                   ? std::optional<SE3>(SE3({point.x, point.y, point.z}, Eigen::Quaterniond::Identity()))
-                   : std::nullopt;
+    if (!std::isfinite(point.x) || !std::isfinite(point.y) || !std::isfinite(point.z)) {
```

Comment:
if this actually happens ever we probably need to do the thing Matt implemented a long time ago which was to check neighboring pixels to see if you can pick up a depth from there.

Reply:
Yeah good idea I will look into that, except I think in practice it usually finds a point.